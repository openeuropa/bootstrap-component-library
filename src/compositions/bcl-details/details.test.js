import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";
import demoData from "@openeuropa/bcl-details/data";

const template = "@oe-bcl/bcl-details/details.html.twig";
const render = (params) => renderTwigFileAsNode(template, params, true);

describe("OE - Details", () => {
  test(`renders correctly`, () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });
});
