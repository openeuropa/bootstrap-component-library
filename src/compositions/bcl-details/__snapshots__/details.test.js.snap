// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Details renders correctly 1`] = `
<jest>
  <h3
    class="fw-bold mb-4"
    id="project-details"
  >
    
  Project details

  </h3>
  <div
    class="border-bottom pb-4 mb-4"
  >
    <div
      class="row"
    >
      <div
        class="col-12 col-md-3"
      >
        <p
          class="fw-bold"
        >
          Status
        </p>
        <span
          class="mb-3 mb-md-0 badge bg-info"
        >
          Ongoing
        </span>
      </div>
      <div
        class="col-12 col-md-9"
      >
        <div
          class="d-flex justify-content-between"
        >
          <p
            class="fw-bold"
          >
            
      Start 
            <br />
            <span
              class="fw-normal"
            >
              19/12/2020
            </span>
          </p>
          <p
            class="fw-bold text-end"
          >
            
      End 
            <br />
            <span
              class="fw-normal"
            >
              19/12/2027
            </span>
          </p>
        </div>
        <div
          class="progress"
        >
          <div
            aria-label="25"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="25"
            class="progress-bar bg-info"
            role="progressbar"
            style="width: 25%"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="border-bottom mb-4"
  >
    <dl
      class="d-md-grid grid-3-9"
    >
      <dt>
        
          Website
      </dt>
      <div>
        <dd>
          <a
            href="/example.html"
          >
            www.website-address.eu
            <svg
              class="ms-2-5 bi icon--fluid"
            >
              <use
                xlink:href="/icons.svg#box-arrow-up-right"
              />
            </svg>
          </a>
        </dd>
      </div>
      <dt>
        
          Funding Programme
      </dt>
      <div>
        <dd>
          <a
            href="/example.html"
          >
            www.funding-prog.eu
            <svg
              class="ms-2-5 bi icon--fluid"
            >
              <use
                xlink:href="/icons.svg#box-arrow-up-right"
              />
            </svg>
          </a>
        </dd>
      </div>
      <dt>
        
          Project ID
      </dt>
      <div>
        <dd>
          <a
            href="/example.html"
          >
            345984678
            <svg
              class="ms-2-5 bi icon--fluid"
            >
              <use
                xlink:href="/icons.svg#box-arrow-up-right"
              />
            </svg>
          </a>
        </dd>
      </div>
    </dl>
  </div>
  <dl
    class="d-md-grid grid-3-9"
  >
    <dt>
      <p
        class="mb-0 d-inline-block"
      >
        Coordinator
      </p>
    </dt>
    <div>
      <dd>
        <svg
          class="me-2 bi icon--fluid"
        >
          <use
            xlink:href="/flags.svg#bulgaria"
          />
        </svg>
        Bulgaria
      </dd>
    </div>
  </dl>
  <h3
    class="fw-bold mb-4 pt-3"
    id="summary"
  >
    
      Summary
    
  </h3>
  <p>
    
        This is the first item's accordion body. It is hidden by default, until the collapse plugin adds the appropriate classes that we use to style each element. These classes control the overall appearance, as well as the showing and hiding via CSS transitions.
      
  </p>
  <p>
    
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse volutpat ultricies massa, a dapibus libero porta nec. Sed facilisis dictum vestibulum. Fusce commodo hendrerit diam, pretium tempus leo varius sit amet. Etiam interdum, orci at sagittis luctus, lorem libero tempus mauris, a fermentum libero orci semper lacus. Duis tristique fringilla magna, eu egestas dolor molestie non. 
      
  </p>
  <h3
    class="fw-bold mb-4 pt-3"
    id="objective"
  >
    
      Objective
    
  </h3>
  <p>
    
        This is the first item's accordion body. It is hidden by default, until the collapse plugin adds the appropriate classes that we use to style each element. These classes control the overall appearance, as well as the showing and hiding via CSS transitions.
      
  </p>
  <p>
    
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse volutpat ultricies massa, a dapibus libero porta nec. Sed facilisis dictum vestibulum. Fusce commodo hendrerit diam, pretium tempus leo varius sit amet. Etiam interdum, orci at sagittis luctus, lorem libero tempus mauris, a fermentum libero orci semper lacus. Duis tristique fringilla magna, eu egestas dolor molestie non. 
      
  </p>
  <h3
    class="fw-bold mb-4 pt-3"
    id="impact"
  >
    
      Impact
    
  </h3>
  <p>
    
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse volutpat ultricies massa, a dapibus libero porta nec. Sed facilisis dictum vestibulum. Fusce commodo hendrerit diam, pretium tempus leo varius sit amet. Etiam interdum, orci at sagittis luctus, lorem libero tempus mauris, a fermentum libero orci semper lacus. Duis tristique fringilla magna, eu egestas dolor molestie non. 
      
  </p>
  <h3
    class="fw-bold mb-4 pt-2"
    id="contributors"
  >
    
    Contributors
  
  </h3>
  <h4
    class="fw-bold mb-3"
  >
    HIT Manufacturing sas
  </h4>
  <dl
    class="border-bottom d-md-grid grid-3-9"
  >
    <dt>
      <p
        class="mb-0 d-inline-block"
      >
        Name
      </p>
    </dt>
    <div>
      <dd>
        Commissariat à l’Energie Atomique 
        <br />
        et aux Energies Alternatives
      </dd>
    </div>
    <dt>
      <p
        class="mb-0 d-inline-block"
      >
        Country
      </p>
    </dt>
    <div>
      <dd>
        <svg
          class="me-2 bi icon--fluid"
        >
          <use
            xlink:href="/flags.svg#france"
          />
        </svg>
        France
      </dd>
    </div>
    <dt>
      <p
        class="mb-0 d-inline-block"
      >
        Contribution
      </p>
    </dt>
    <div>
      <dd>
        EUR 297.700.00
      </dd>
    </div>
  </dl>
  <h4
    class="fw-bold mb-3"
  >
    Multi-informatica principado
  </h4>
  <dl
    class="border-bottom d-md-grid grid-3-9"
  >
    <dt>
      <p
        class="mb-0 d-inline-block"
      >
        Name
      </p>
    </dt>
    <div>
      <dd>
        Lorem Ipsum Color Sit Amet
      </dd>
    </div>
    <dt>
      <p
        class="mb-0 d-inline-block"
      >
        Country
      </p>
    </dt>
    <div>
      <dd>
        <svg
          class="me-2 bi icon--fluid"
        >
          <use
            xlink:href="/flags.svg#bulgaria"
          />
        </svg>
        Bulgaria
      </dd>
    </div>
    <dt>
      <p
        class="mb-0 d-inline-block"
      >
        Contribution
      </p>
    </dt>
    <div>
      <dd>
        EUR 297.700.00
      </dd>
    </div>
  </dl>
  <dl
    class="border-bottom d-md-grid grid-3-9"
  >
    <dt>
      <p
        class="mb-0 d-inline-block"
      >
        Name
      </p>
    </dt>
    <div>
      <dd>
        Lorem Ipsum Color Sit Amet
      </dd>
    </div>
    <dt>
      <p
        class="mb-0 d-inline-block"
      >
        Country
      </p>
    </dt>
    <div>
      <dd>
        <svg
          class="me-2 bi icon--fluid"
        >
          <use
            xlink:href="/flags.svg#belgium"
          />
        </svg>
        Belgium
      </dd>
    </div>
    <dt>
      <p
        class="mb-0 d-inline-block"
      >
        Contribution
      </p>
    </dt>
    <div>
      <dd>
        EUR 297.700.00
      </dd>
    </div>
  </dl>
  <dl
    class="d-md-grid grid-3-9"
  >
    <dt>
      <p
        class="mb-0 d-inline-block"
      >
        Name
      </p>
    </dt>
    <div>
      <dd>
        Lorem Ipsum Color Sit Amet
      </dd>
    </div>
    <dt>
      <p
        class="mb-0 d-inline-block"
      >
        Country
      </p>
    </dt>
    <div>
      <dd>
        <svg
          class="me-2 bi icon--fluid"
        >
          <use
            xlink:href="/flags.svg#denmark"
          />
        </svg>
        Denmark
      </dd>
    </div>
    <dt>
      <p
        class="mb-0 d-inline-block"
      >
        Contribution
      </p>
    </dt>
    <div>
      <dd>
        EUR 297.700.00
      </dd>
    </div>
  </dl>
  <h3
    class="fw-bold mb-4 pt-2"
    id="achievements"
  >
    
        Achievements and Milestones
      
  </h3>
  <p>
    
        This is the first item's accordion body. It is hidden by default, until the collapse plugin adds the appropriate classes that we use to style each element. These classes control the overall appearance, as well as the showing and hiding via CSS transitions.
      
  </p>
  <p>
    
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse volutpat ultricies massa, a dapibus libero porta nec. Sed facilisis dictum vestibulum. Fusce commodo hendrerit diam, pretium tempus leo varius sit amet. Etiam interdum, orci at sagittis luctus, lorem libero tempus mauris, a fermentum libero orci semper lacus. Duis tristique fringilla magna, eu egestas dolor molestie non. 
      
  </p>
  <h3
    class="fw-bold mb-4 pt-2"
    id="gallery"
  >
    
      Gallery
    
  </h3>
  <figure
    class="rounded overflow-hidden"
  >
    <video
      class="w-100 d-block"
      controls=""
      poster="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
    >
      <source
        src="https://inno-ecl.s3.amazonaws.com/media/videos/big_buck_bunny.mp4"
        type="video/mp4"
      />
      <source
        src="https://inno-ecl.s3.amazonaws.com/media/videos/big_buck_bunny.webm"
        type="video/webm"
      />
      <track
        kind="captions"
        label="English"
        src="example.vtt"
        srclang=""
      />
    </video>
  </figure>
  <div
    class="carousel slide"
    data-bs-ride="carousel"
    id="carouselExample"
  >
    <div
      class="carousel-indicators"
    >
      <button
        aria-current="true"
        aria-label="Slide 1"
        class="active"
        data-bs-slide-to="0"
        data-bs-target="#carouselExample"
        type="button"
      />
      <button
        aria-label="Slide 2"
        data-bs-slide-to="1"
        data-bs-target="#carouselExample"
        type="button"
      />
      <button
        aria-label="Slide 3"
        data-bs-slide-to="2"
        data-bs-target="#carouselExample"
        type="button"
      />
    </div>
    <div
      class="carousel-inner"
    >
      <div
        class="carousel-item active"
      >
        <img
          alt="First slide"
          class="d-block w-100"
          src="https://picsum.photos/800/400?random=1"
        />
        <div
          class="carousel-caption d-none d-md-block"
        >
          <h5>
            First slide label
          </h5>
          <p>
            Some representative placeholder content for the first slide.
          </p>
        </div>
      </div>
      <div
        class="carousel-item"
      >
        <img
          alt="Second slide"
          class="d-block w-100"
          src="https://picsum.photos/800/400?random=2"
        />
      </div>
      <div
        class="carousel-item"
      >
        <img
          class="d-block w-100"
          src="https://picsum.photos/800/400?random=3"
        />
        <div
          class="carousel-caption d-none d-md-block"
        >
          <h5>
            Third slide label
          </h5>
          <p>
            Some representative placeholder content for the third slide.
          </p>
        </div>
      </div>
    </div>
    <button
      class="carousel-control-prev"
      data-bs-slide="prev"
      data-bs-target="#carouselExample"
      type="button"
    >
      <span
        aria-hidden="true"
        class="carousel-control-prev-icon"
      />
      <span
        class="visually-hidden"
      >
        Previous
      </span>
    </button>
    <button
      class="carousel-control-next"
      data-bs-slide="next"
      data-bs-target="#carouselExample"
      type="button"
    >
      <span
        aria-hidden="true"
        class="carousel-control-next-icon"
      />
      <span
        class="visually-hidden"
      >
        Next
      </span>
    </button>
  </div>
</jest>
`;
